crates/deltalake-core/src/operations/transaction/protocol.rs

```diff
@@ -170,7 +170,7 @@ pub static INSTANCE: Lazy<ProtocolChecker> = Lazy::new(|| {
     let mut writer_features = HashSet::new();
     writer_features.insert(WriterFeatures::AppendOnly);
     writer_features.insert(WriterFeatures::Invariants);
-    // writer_features.insert(WriterFeatures::CheckConstraints);
+    writer_features.insert(WriterFeatures::CheckConstraints);
```

Comment:
@hntd187 @scarman-db - one last thing, could we hide `Invariants` and `CheckConstraints` behind the `datafusion` features, as they will not work otherwise ...